### ISSUE TITLE
Suggestion: keep hyperparameters for all iterations

### ISSUE THREAD

**Issue by torronen** [dotnet/machinelearning-modelbuilder#1969]
**Is your feature request related to a problem? Please describe.**
I have dataset split into training and test set. I am using train set with Model builder. 
I have been now testing all the models in temp folder against the train set (not just the generated one but the 1....999.zip's in temp)
The best model selected for model builder is not always the best in test data set.

**Describe the solution you'd like**
I'd like to be able to retrieve the hyperparameters used for training of each zip file in temp file. 
It could be, for examples, in .mbconfig file. Saving them would also help later restart the training with different kinds of tuning engines, based on the results.

**Describe alternatives you've considered**
Best solution could be that the hyperparameters would be saved inside the .zip file, because retraining is not possible without it for many algorithms (such as trees). That would make archiving models easier, as there would not need to be a separate parameters file.

**Comment by torronen** [comment]
Adding the parameters to log file would probably be also adequate. For special cases, like mine, I could check log file. It might also help in debugging issues. Currently, log file does not include much data. 
Could it be considered:
 -  the parameters would be saved here. 
 - all trainers would be set to verbose, so we could see if there are any warnings about data quality

**Comment by beccamc** [comment]
Thanks @torronen! This is great feedback

**Comment by torronen** [comment]
If you will consider additional logging, maybe same format as FLAML? It seems to have some tooling for plotting the log file.

**Comment by torronen** [comment]
see https://github.com/dotnet/machinelearning-modelbuilder/issues/1999#issuecomment-1030697354
